cmd/containerboot/main.go

```diff
@@ -309,7 +314,7 @@ authLoop:
 		}
 	}
 
-	if cfg.InKubernetes && cfg.KubeSecret != "" && cfg.KubernetesCanPatch && cfg.AuthOnce {
+	if cfg.InKubernetes && cfg.KubeSecret != "" && cfg.KubernetesCanPatch && runTailscaleSet(cfg) {
```

Comment:
Again, weird to see `runTailscaleSet` gating something other than tailscaleSet. Perhaps we could use a better function name, or gate this by AuthOnce and runTailscaledOnly like you do on line 172?

Reply:
Same as in https://github.com/tailscale/tailscale/pull/10759#discussion_r1444919730